src/Servers/Kestrel/samples/Http3SampleApp/Program.cs

```diff
@@ -35,6 +35,13 @@ public static void Main(string[] args)
                         listenOptions.Protocols = HttpProtocols.Http1AndHttp2;
                     });
 
+                    // On a machine that supports IPv6, msquic will resolve `localhost` to `::1`, which `IPAddress.Any` will not accept.
+                    // (To make it work, you'd need to use `IPv6Any`, as `ListenAnyIP` does.)  As a result, a client built on msquic, 
+                    // e.g. `HttpClient` on Windows, will not be able to make HTTP/3.0 requests to `localhost:5001`.  HTTP/1.x and 
+                    // HTTP/2.0 will still work.
+                    // Example: https://github.com/dotnet/runtime/issues/108259
+                    // msquic: https://github.com/microsoft/msquic/issues/1181 (resolve both)
+                    // System.Net.Quic: https://github.com/dotnet/runtime/issues/82404 (improve consistency with sockets)
                     options.Listen(IPAddress.Any, 5001, listenOptions =>
```

Comment:
Should we change this to `IPv6Any`.

Reply:
Depends on the purpose of the sample.  If it's what we think people should do, we should just `ListenAnyIP`.  However, my impression was that it was illustrating different ways of handling things, so I thought it was more useful to exhibit this scenario and make it testable.  I don't feel strongly though.

Reply:
Sticking with just the explanation for now, primarily because CI is green. 😆